src/main/java/org/embulk/util/guess/SchemaGuess.java

```diff
@@ -304,7 +304,7 @@ private GuessedType mergeTypes(final List<GuessedType> types) {
         if (t.isTimestamp()) {
             final String format = this.timeFormatGuess.guess(
                     types.stream()
-                            .map(type -> type.isTimestamp() ? type.getFormatOrTimeValue() : null)
+                            .map(type -> (type != null && type.isTimestamp()) ? type.getFormatOrTimeValue() : null)
```

Comment:
other way, move `filter(type -> type != null)` before `map`  will avoid null check

Reply:
I thought that once, but `.map(...)` can also adds `null` again, and `.filter` is required again. So it would be more straightforward.